packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -70,4 +115,283 @@ class AndroidCameraCameraX extends CameraPlatform {
+  /// [processCameraProvider].
+  Future<void> _bindPreviewToLifecycle() async {
+    assert(processCameraProvider != null);
+    assert(cameraSelector != null);
+
+    if (previewIsBound || _previewIsPaused) {
+      // Only bind if preview is not already bound or intentionally paused.
+      return;
+    }
+
+    camera = await processCameraProvider!
+        .bindToLifecycle(cameraSelector!, <UseCase>[preview!]);
+    previewIsBound = true;
+  }
+
+  /// Unbinds [preview] instance to camera lifecycle controlled by the
+  /// [processCameraProvider].
+  void _unbindPreviewFromLifecycle() {
+    if (preview == null || !previewIsBound) {
+      return;
+    }
+
+    assert(processCameraProvider != null);
+
+    processCameraProvider!.unbind(<UseCase>[preview!]);
+    previewIsBound = false;
+  }
+
+  // Methods for mapping Flutter camera constants to CameraX constants:
+
+  /// Returns [CameraSelector] lens direction that maps to specified
+  /// [CameraLensDirection].
+  int _getCameraSelectorLensDirection(CameraLensDirection lensDirection) {
+    switch (lensDirection) {
+      case CameraLensDirection.front:
+        return CameraSelector.LENS_FACING_FRONT;
+      case CameraLensDirection.back:
+        return CameraSelector.LENS_FACING_BACK;
+      case CameraLensDirection.external:
+        return CameraSelector.EXTERNAL;
+    }
+  }
+
+  /// Returns [Surface] target rotation constant that maps to specified sensor
+  /// orientation.
+  int _getTargetRotation(int sensorOrientation) {
+    switch (sensorOrientation) {
+      case 90:
+        return Surface.ROTATION_90;
+      case 180:
+        return Surface.ROTATION_180;
+      case 270:
+        return Surface.ROTATION_270;
+      case 0:
+        return Surface.ROTATION_0;
+      default:
+        throw ArgumentError(
+            '"$sensorOrientation" is not a valid sensor orientation value');
+    }
+  }
+
+  /// Returns [ResolutionInfo] that maps to the specified resolution preset for
+  /// a camera preview.
+  ResolutionInfo? _getTargetResolutionForPreview(ResolutionPreset? resolution) {
+    // TODO(camsim99): Implement resolution configuration.
+    // https://github.com/flutter/flutter/issues/120462
+    return null;
+  }
+
+  // Methods for calls that need to be tested:
+
+  /// Requests camera permissions.
+  @visibleForTesting
+  Future<void> requestCameraPermissions(bool enableAudio) async {
+    await SystemServices.requestCameraPermissions(enableAudio);
+  }
+
+  /// Subscribes the plugin as a listener to changes in device orientation.
+  @visibleForTesting
+  void startListeningForDeviceOrientationChange(
+      bool cameraIsFrontFacing, int sensorOrientation) {
+    SystemServices.startListeningForDeviceOrientationChange(
+        cameraIsFrontFacing, sensorOrientation);
+  }
+
+  /// Retrives an instance of the [ProcessCameraProvider] to access camera
+  /// functionality.
+  @visibleForTesting
+  Future<ProcessCameraProvider> getProcessCameraProviderInstance() async {
```

Comment:
Is this method necessary if you already expose the `processCameraProvider` variable for testing?